servarr/config/qbittorrent/init-qBitTorrent.py

```diff
@@ -0,0 +1,164 @@
+QBITTORRENT_CONF_FILEPATH = "/mnt/qBittorrent/config"
+QBITTORRENT_CONF_TEMPLATE = r"""
+[Application]
+FileLogger\Age=1
+FileLogger\AgeType=1
+FileLogger\Backup=true
+FileLogger\DeleteOld=true
+FileLogger\Enabled=true
+FileLogger\MaxSizeBytes=66560
+FileLogger\Path=/config/qBittorrent/logs
+
+[BitTorrent]
+Session\DefaultSavePath=/downloads
+Session\ExcludedFileNames=
+Session\Port=50413
+Session\QueueingSystemEnabled=true
+Session\TempPath=/downloads/temp
+
+[Core]
+AutoDeleteAddedTorrentFile=Never
+
+[LegalNotice]
+Accepted=true
+
+[Meta]
+MigrationVersion=6
+
+[Network]
+Proxy\HostnameLookupEnabled=false
+Proxy\Profiles\BitTorrent=true
+Proxy\Profiles\Misc=true
+Proxy\Profiles\RSS=true
+
+[Preferences]
+General\Locale=en
+MailNotification\req_auth=true
+WebUI\AuthSubnetWhitelist=192.168.0.0/24, 10.1.0.0/24
+WebUI\AuthSubnetWhitelistEnabled=true
+WebUI\HostHeaderValidation=false
+WebUI\LocalHostAuth=false
+
+#### UPDATE this value
+WebUI\Password_PBKDF2="{{`{{ torrentPassword }}`}}"
+WebUI\UseUPnP=false
+WebUI\Username={{`{{ torrentUsername }}`}}
+
+[RSS]
+AutoDownloader\DownloadRepacks=true
+"""
+
+
+def create_file(file_content: str):
+    file_absolute_path = QBITTORRENT_CONF_FILEPATH + os.sep + QBITTORRENT_CONF_FILENAME
+    os.umask(0o077)
+    with open(file_absolute_path, "w+") as fd:
+        fd.write(file_content)
+    return file_absolute_path
+
+
+def qbittorrent_passwd(plain_passwd: str):
+    # As per https://github.com/qbittorrent/qBittorrent/blob/ce9bdaef5cdb8ab77d71481f20b25c9e6da1b9eb/src/base/utils/password.cpp#L48
+    ITERATIONS = 100_000
+    # 4x32 bits words = 16 bytes: https://github.com/qbittorrent/qBittorrent/blob/ce9bdaef5cdb8ab77d71481f20b25c9e6da1b9eb/src/base/utils/password.cpp#L75
+    SALT_SIZE = 16
+    # Generate a cryptographically secure pseudorandom salt
+    salt = os.urandom(SALT_SIZE)
+    # PBKDF2 w/ SHA512 hmac
+    h = hashlib.pbkdf2_hmac("sha512", plain_passwd.encode(), salt, ITERATIONS)
+    # Base64 encode and join salt and hash
+    return (
+        f"@ByteArray({base64.b64encode(salt).decode()}:{base64.b64encode(h).decode()})"
+    )
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+console_handler = logging.StreamHandler()
+log_format = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+console_handler.setFormatter(log_format)
+logger.addHandler(console_handler)
+
+TORRENT_USERNAME = os.getenv("TORRENT_USERNAME", "admin")
+TORRENT_PASSWORD = os.getenv("TORRENT_PASSWORD", "password")
+
+logger.info("qBitTorrent Init Job started")
+logger.info("The Job will configure qBitTorrent credentials")
+
+logger.info("Checking credentials")
+if TORRENT_USERNAME in [None, ""]:
+    logger.error("Empty username passed")
+    sys.exit(1)
+if TORRENT_PASSWORD in [None, ""]:
+    logger.error("Empty password passed")
+    sys.exit(1)
+if len(TORRENT_PASSWORD) < MIN_PASS_LEN:
+    logger.error("Password too short, it should be at least of 8 characters")
```

Comment:
Should we update the `values.yaml` documentation and add the password length constraint?

https://github.com/fonzdm/servarr/blob/51cce1effa3900332d1b85f3661375d632dca5d4/servarr/values.yaml#L55

Reply:
Right!
Done 👍🏻